pybamm/models/full_battery_models/lithium_ion/basic_dfn.py

```diff
@@ -100,9 +100,15 @@ def __init__(self, name="Doyle-Fuller-Newman model"):
         # Porosity
         # Primary broadcasts are used to broadcast scalar quantities across a domain
         # into a vector of the right shape, for multiplying with other vectors
-        eps_n = pybamm.PrimaryBroadcast(param.epsilon_n, "negative electrode")
-        eps_s = pybamm.PrimaryBroadcast(param.epsilon_s, "separator")
-        eps_p = pybamm.PrimaryBroadcast(param.epsilon_p, "positive electrode")
+        eps_n = pybamm.PrimaryBroadcast(
+            pybamm.Parameter("Negative electrode porosity"), "negative electrode"
```

Comment:
I agree with Scott. It would be good to pass it as a function or dataset, as with other parameters.

Reply:
yeah, the problem is that `param.epsilon_s` now has domain "separator" and secondary domain "current collector", and so we would need to add "current collector" as a secondary domain to everything to avoid shape errors (tbh this should probably be automatic), and adding current collector to everything is unnecessarily complicated for this model